src/ocp_postprocess/hostname_rename/etcd_rename.rs

```diff
@@ -387,6 +387,43 @@ pub(crate) async fn fix_etcds_cluster(etcd_client: &Arc<InMemoryK8sEtcd>, hostna
     Ok(())
 }
 
+pub(crate) async fn fix_node_name(etcd_client: &Arc<InMemoryK8sEtcd>, original_hostname: &str, hostname: &str) -> Result<()> {
+    let mut k8s_resource_location = K8sResourceLocation::new(None, "Minion", original_hostname, "v1");
+    let mut node = get_etcd_json(etcd_client, &k8s_resource_location)
+        .await?
+        .context(format!("getting node {original_hostname}"))?;
+
+    let metadata = &mut node
+        .pointer_mut("/metadata")
+        .context("no metadata")?
+        .as_object_mut()
+        .context("/metadata not an object")?;
+
+    metadata.insert("name".to_string(), Value::String(hostname.to_string()));
+
+    let addresses = &mut node
+        .pointer_mut("/status/addresses")
+        .context("no /status/addresses")?
+        .as_array_mut()
+        .context("/status/addresses not an array")?;
+
+    addresses
+        .iter_mut()
+        .filter(|address| address["type"] == "Hostname")
+        .for_each(|address: &mut Value| {
+            address
+                .as_object_mut()
+                .unwrap()
```

Comment:
`unwrap()` on an `Option` that has the potential to be `None` can cause panics, and recert generally never panics, so this should be avoided.

Instead use `anyhow`'s `.context("something")` to convert the `Option<>` type into a `Result<>` (if the `Option<>` is `None` the resulting `Result` will be an `Err`, otherwise if the `Option` is a `Some(_)` the resulting `Result` will be an `Ok(_)` containing the same value as the original `Option`. After this you can `?` to return an error on `Err` from the `for_each`'s lambda, which you should change to a `try_for_each` instead, which would return the first error upon the first error, and wrap the entire `address. ... . insert` expression in an `Ok`, then you can `?` on the `try_for_each`'s return value

Reply:
Awesome, I'll try to incorporate everything you just explained :pray:

Reply:
Updated PTAL (although we probably won't keep this code, as the status should be updated by the respective controller without any issues)

Reply:
Ah lol didn't notice it's part of the status

Reply:
:D absolutely no worries TIL :muscle: